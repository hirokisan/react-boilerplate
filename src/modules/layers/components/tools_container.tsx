import React, { FC } from 'react';

import { classes, Action, stringAttribute, theme, styled } from 'config/common';

const ToolsList = styled.ul`
  width: 100%;
  height: 40px;
  min-height: 40px;
  align-items: center;
  border: 1px solid ${theme.divider};
  display: flex;
  justify-content: flex-start;
`;
ToolsList.displayName = 'ToolsList';

const ToolItem = styled.li`
  width: 40px;
  height: 30px;
  align-items: center;
  border: 1px solid ${theme.divider};
  border-radius: 2px;
  cursor: pointer;
  display: flex;
  justify-content: center;
  margin-left: 10px;
  &:hover,
  &.active {
    background-color: ${theme.paneActive};
  }
  .icon {
    width: 18px;
  }
`;
ToolItem.displayName = 'ToolItem';

// const ToolIcon = styled.img`
//   width: 18px;
// `;

const ToolIcon = styled.img`
  width: 18px;
  label: Icon;
`;
ToolIcon.displayName = 'ToolIcon';

type Props = {
  setTool: Action<string>;
};

const tools = [
  {
    name: 'select',
    icon: 'https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg'
  },
  {
    name: 'edit',
    icon: 'https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg'
  },
  {
    name: 'measure',
    icon: 'https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg'
  }
];

export const ToolsContainer: FC<Props> = ({ setTool }) => {
  const [tool, setToolActive] = React.useState('select');
  const setToolHandler = React.useCallback(
    (e: React.MouseEvent<HTMLLIElement>) => {
      const tool = stringAttribute(e.currentTarget, 'data-tool');
      setToolActive(tool);
      setTool(tool);
    },
    [setTool]
  );

  return (
    <ToolsList>
      {tools.map(t => (
        <ToolItem
          key={t.name}
          className={classes({ active: tool === t.name })}
          data-tool={t.name}
          onClick={setToolHandler}
        >
          <ToolIcon src={t.icon} />
        </ToolItem>
      ))}
    </ToolsList>
  );
};
