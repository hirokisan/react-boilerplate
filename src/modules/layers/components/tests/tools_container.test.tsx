import React from 'react';
import renderer from 'react-test-renderer';

import { ToolsContainer } from '../tools_container';

describe('Tools View', () => {
  const setTool = jest.fn();
  const tree = renderer.create(<ToolsContainer setTool={setTool} />);

  it('renders default', () => {
    expect(tree).toMatchSnapshot();
  });

  it('selects another item and calls a callback function', () => {
    const fakeEvent = {
      currentTarget: {
        hasAttribute: () => true,
        getAttribute: () => 'edit'
      }
    };
    tree.root.findByProps({ 'data-tool': 'edit' }).props.onClick(fakeEvent);
    expect(tree).toMatchSnapshot();
    expect(setTool).toHaveBeenCalledWith('edit');
  });
});
