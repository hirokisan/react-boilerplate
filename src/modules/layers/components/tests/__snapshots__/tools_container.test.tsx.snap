// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tools View calls the callback function 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii active"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View renders default 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii active"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item a 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and  1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and ca 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and cal 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and call 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and calls  1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and calls a  1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and calls a c 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and calls a ca 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and calls a cal 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and calls a callba 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and calls a callback  1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and calls a callback f 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and calls a callback fu 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and calls a callback func 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and calls a callback functio 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;

exports[`Tools View selects another item and calls a callback function 1`] = `
<ul
  className="css-106gbz6"
>
  <li
    className="css-xkmlii"
    data-tool="select"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
    />
  </li>
  <li
    className="css-xkmlii active"
    data-tool="edit"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
    />
  </li>
  <li
    className="css-xkmlii"
    data-tool="measure"
    onClick={[Function]}
  >
    <img
      className="css-1ewz7j9-Icon"
      src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
    />
  </li>
</ul>
`;
